imcache.go

```diff
@@ -249,6 +249,38 @@ func (c *Cache[K, V]) getAll(now time.Time) map[K]V {
 	return got
 }
 
+// GetAndRemove returns the value for the given key
+// and delete it even if it is not expired
```

Comment:
```suggestion
// GetAndRemove returns the value for the given key and removes it if it
// exists, otherwise it returns zero value and false.
//
// If it encounters an expired entry, the expired entry is evicted.
```